python/langsmith/client.py

```diff
@@ -931,6 +932,43 @@ def request_with_retries(
                     filler = "*" * (max(0, len(api_key) - 7))
                     masked_api_key = f"{prefix}{filler}{suffix}"
 
+                    debug_crash_dump_file = ls_utils.get_env_var(
+                        "LANGSMITH_DEBUG_CRASH_DUMP"
+                    )
+                    if debug_crash_dump_file is not None and e.request is not None:
+                        try:
+                            headers = e.request.headers.copy()
+                            headers.pop("x-api-key", None)
+                            request_to_log = {
+                                "method": e.request.method,
+                                "url": e.request.url,
+                                "headers": headers,
+                                "body": (
+                                    e.request.body
+                                    if isinstance(e.request, requests.PreparedRequest)
+                                    else None
+                                ),
+                                "data": (
+                                    e.request.data
+                                    if isinstance(e.request, requests.Request)
+                                    else None
+                                ),
+                                "json": (
+                                    e.request.json
+                                    if isinstance(e.request, requests.Request)
+                                    else None
+                                ),
+                            }
+                            with gzip.open(
```

Comment:
can you easily open a gzip file that has been appended to multiple times?

Reply:
Just looked into it a bit, it is a bit of a pain to read from multiple streams.

Do you think an acceptable alternative is to write to a file with a certain timestamp appended to the file path?

We can probably turn the env variable back to a boolean, and then use a fixed static name like `debug_crash_dump_{timestamp}`. wdyt? @nfcampos